File: 20221013/모범답안/20221013_08/PercentageOfLetter_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.medium.PercentageOfLetter.PercentageOfLetter import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        text = 'google'
        s = 'o'

        # when
        result = solution(text, s)

        # then
        self.assertEqual(33, result)

    def test_2(self):
        # given
        text = 'naver'
        s = 'o'

        # when
        result = solution(text, s)

        # then
        self.assertEqual(0, result)

    def test_3(self):
        # given
        text = 'a'
        s = 'a'

        # when
        result = solution(text, s)

        # then
        self.assertEqual(100, result)

    def test_4(self):
        # given
        text = 'nrttxlpujq'
        s = 'd'

        # when
        result = solution(text, s)

        # then
        self.assertEqual(0, result)

    def test_5(self):
        # given
        text = 'bnwbjmyzagosnvljorpi'
        s = 'a'

        # when
        result = solution(text, s)

        # then
        self.assertEqual(5, result)

    def test_6(self):
        # given
        text = 'okvyyvahyhtdyffoaeezoyvmczeekftvjlfuwoybxstmjkqvykoflvoknhmxjfjdphgkspczaearxiojdisadurghlbtfzlbqcds'
        s = 'v'

        # when
        result = solution(text, s)

        # then
        self.assertEqual(6, result)


if __name__ == '__main__':
    unittest.main()

File: 20221012/모범답안/20221012_08/PercentageOfLetter_test.js
import assert from 'assert'
import solution from './PercentageOfLetter.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const text = 'google'
    const s = 'o'

    // when
    const result = solution(text, s)

    // then
    assert.deepStrictEqual(33, result)
    done()
  })

  it('test 2', function (done) {
    // given
    const text = 'naver'
    const s = 'o'

    // when
    const result = solution(text, s)

    // then
    assert.deepStrictEqual(0, result)
    done()
  })

  it('test 3', function (done) {
    // given
    const text = 'a'
    const s = 'a'

    // when
    const result = solution(text, s)

    // then
    assert.deepStrictEqual(100, result)
    done()
  })

  it('test 4', function (done) {
    // given
    const text = 'nrttxlpujq'
    const s = 'd'

    // when
    const result = solution(text, s)

    // then
    assert.deepStrictEqual(0, result)
    done()
  })

  it('test 5', function (done) {
    // given
    const text = 'bnwbjmyzagosnvljorpi'
    const s = 'a'

    // when
    const result = solution(text, s)

    // then
    assert.deepStrictEqual(5, result)
    done()
  })

  it('test 6', function (done) {
    // given
    const text = 'okvyyvahyhtdyffoaeezoyvmczeekftvjlfuwoybxstmjkqvykoflvoknhmxjfjdphgkspczaearxiojdisadurghlbtfzlbqcds'
    const s = 'v'

    // when
    const result = solution(text, s)

    // then
    assert.deepStrictEqual(6, result)
    done()
  })
})
